packages/license-scanner/src/licenses.ts

```diff
@@ -0,0 +1,128 @@
+import { PnpmError } from '@pnpm/error'
+import { Lockfile, PackageSnapshot } from '@pnpm/lockfile-file'
+import {
+  DependenciesField,
+  IncludedDependencies,
+  ProjectManifest,
+  Registries,
+} from '@pnpm/types'
+import { getPkgInfo } from './getPkgInfo'
+import {
+  LicenseNode,
+  lockfileToLicenseNodeTree,
+} from './lockfileToLicenseNodeTree'
+
+export interface LicensePackage {
+  belongsTo: DependenciesField
+  version: string
+  name: string
+  license: string
+  licenseContents?: string
+  author?: string
+  homepage?: string
+  repository?: string
+  path?: string
+}
+
+export type GetPackageInfoFunction = (
+  pkg: {
+    name?: string
+    version?: string
+    depPath: string
+    snapshot: PackageSnapshot
+    registries: Registries
+  },
+  opts: {
+    storeDir: string
+    virtualStoreDir: string
+    dir: string
+    modulesDir: string
+  }
+) => Promise<{
+  from: string
+  description?: string
+} & Omit<LicensePackage, 'belongsTo'>>
+
+/**
+ * @private
+ * Returns an array of LicensePackages from the given LicenseNode
+ * @param licenseNode the license node
+ * @returns LicensePackage[]
+ */
+function getDependenciesFromLicenseNode (
+  licenseNode: LicenseNode
+): LicensePackage[] {
+  if (!licenseNode.dependencies) {
+    return []
+  }
+
+  let dependencies: LicensePackage[] = []
+  for (const dependencyName in licenseNode.dependencies) {
+    const dependencyNode = licenseNode.dependencies[dependencyName]
+    const dependenciesOfNode = getDependenciesFromLicenseNode(dependencyNode)
+
+    dependencies = [
+      ...dependencies,
+      ...dependenciesOfNode,
+      {
+        belongsTo: dependencyNode.dev ? 'devDependencies' : 'dependencies',
+        version: dependencyNode.version as string,
+        name: dependencyName,
+        license: dependencyNode.license as string,
+        licenseContents: dependencyNode.licenseContents,
+        author: dependencyNode.author as string,
+        homepage: dependencyNode.homepage as string,
+        repository: dependencyNode.repository as string,
+        path: dependencyNode.dir,
+      },
+    ]
+  }
+
+  return dependencies
+}
+
+export async function findDependencyLicenses (opts: {
+  getPackageInfo?: GetPackageInfoFunction
```

Comment:
Don't make this function public. The test is not worth it. This way any breaking change to the internal function will require a major version bump.